tests/test_transforms.py

```diff
@@ -172,18 +178,297 @@ def test_transformed_logprob(at_dist, dist_params, sp_dist, size):
 
-        a_trans_value = a_forward_fn(a_val)
         logprob_val = logp_vals_fn(a_trans_value, b_val)
 
         np.testing.assert_almost_equal(exp_logprob_val, logprob_val, decimal=decimals)
+
+
+def test_default_log_jac_det():
+    class SquareTransform(RVTransform):
+        name = "square"
+
+        def forward(self, value, *inputs):
+            return at.power(value, 2)
+
+        def backward(self, value, *inputs):
+            return at.sqrt(value)
+
+    square_tr = SquareTransform()
+
+    value = at.scalar("value")
+    value_tr = square_tr.forward(value)
+    log_jac_det = square_tr.log_jac_det(value_tr)
+
+    assert np.isclose(log_jac_det.eval({value: 3}), -np.log(6))
+
+
+def test_no_transforms():
+    """
+    Test that joint_logprob succeeds when transform_opt is set, but graph has no
+    transformable variables.
+    """
+    a = at.random.poisson(1)
+    a.name = "a"
+    a_value_var = a.clone()
+    a_value_var.name = "a_value"
+
+    b = at.random.poisson(a)
+    b.name = "b"
+    b_value_var = b.clone()
+    b_value_var.name = "b_value"
+
+    transform_opt = in2out(default_backward_transform, ignore_newtrees=True)
+    res = joint_logprob(
+        b, {a: a_value_var, b: b_value_var}, extra_rewrites=transform_opt
+    )
+
+    logp_vals_fn = aesara.function([a_value_var, b_value_var], res)
+    assert np.isclose(logp_vals_fn(1, 1), sp.stats.poisson(1).logpmf(1) * 2)
+
+
+def test_nondefault_transform():
+    """
+    Test that non default transforms do not overwrite default ones
+    """
+    LogoddsNormalRV = create_transformed_rv_op(
+        at.random.normal, LogOddsTransform(), default=False
+    )()
+
+    a = LogoddsNormalRV.make_node(*at.random.normal(0, 1).owner.inputs).default_output()
+    a.name = "a"
+    a_value_var = a.clone()
+    a_value_var.name = "a_value"
+
+    b = at.random.normal(a, 1)
+    b.name = "b"
+    b_value_var = b.clone()
+    b_value_var.name = "b_value"
+
+    transform_opt = in2out(default_backward_transform, ignore_newtrees=True)
+    res = joint_logprob(
+        b, {a: a_value_var, b: b_value_var}, extra_rewrites=transform_opt
+    )
+    logp_vals_fn = aesara.function([a_value_var, b_value_var], res)
+
+    for a_val, b_val in itertools.product((0.1, 0.5, 0.8), (-2, 0, 0.5)):
+        # logodds forward transform
+        transf_a_val = np.log(a_val / (1 - a_val))
+
+        logp_term_a = sp.stats.norm(0).logpdf(a_val)
+        # log jacobian of logodds  = log(expit(a) x (1 - expit(a))
+        expit_a = sp.special.expit(transf_a_val)
+        log_jacob_term_a = np.log(expit_a) + np.log1p(-expit_a)
+
+        logp_term_b = sp.stats.norm(a_val).logpdf(b_val)
+
+        total_logp = logp_term_a + log_jacob_term_a + logp_term_b
+        assert np.isclose(logp_vals_fn(transf_a_val, b_val), total_logp)
+
+
+@pytest.mark.parametrize("scalar_loc", (True, False))
+@pytest.mark.parametrize("rv_size", [None, 1, 2, (3, 2)])
+def test_loc_transform(scalar_loc, rv_size):
```

Comment:
These tests fail only for the combination: `scalar_loc == False` and `rv_size==1`.

Reply:
The derived TransformedRV has type `TensorType(float64, (True,))`, while the original RV had type `TensorType(float64, vector)`.

Reply:
Is there something special about `size=1`?

Reply:
Seems weird to have to force the `loc` or `scale` variables to be of the same `TensorType` just for this case.

Reply:
See https://github.com/aesara-devs/aeppl/pull/26#issuecomment-900945477 for more context